hubblestack/extmods/fdg/stat.py

```diff
@@ -0,0 +1,250 @@
+       check_stats
+
+check_stats:                    # this profile's check_stats module
+    module: stat.check_stats   # this tells FDG to call stat submodule's check_stats function
+    args:
+        - params:
+            mode: 644
+            gid: 0
+            group: root
+            uid: 0
+            user: root
+            allow_more_strict: True
+            match_on_file_missing: True
+
+*****************************************************
+
+If `match_on_file_missing` is omitted, success/failure will be determined
+entirely based on the other arguments. If it's set to True and
+the file is missing, then it will be considered a match (success).
+If it's set to False and the file is missing, then it
+will be considered a non-match (failure).
+If the file exists, this setting is ignored.
+'''
+
+import logging
+import salt.utils
+import salt.utils.platform
+import os
+log = logging.getLogger(__name__)
+
+
+def check_stats(params='', chained=None, chained_status=None):
+    '''
+
+    :param params:
+    :param chained:
+    :param chained_status:
+    :return: tuple with (status(Boolean), result(dict))
+    This function takes into input stat params that are to be matched with stats of given file.
+    The filepath can be provided either directly in params or through chaining. See example above.
+    '''
+    params = params.get('params')
+
+    if chained != None:
+        log.info("value of 'chained' is not null, using {0} value as filepath".format(chained))
+        filepath = chained.get('filepath')
+    else:
+        filepath = params.get('filepath')
+
+    expected = {}
+    for attribute in ['mode', 'user', 'uid', 'group', 'gid', 'allow_more_strict', 'match_on_file_missing']:
+        if attribute in params.keys():
+            expected[attribute] = params[attribute]
+
+    status, ret = _validate_inputs(filepath, expected)
+    if not status:
+        log.info("Invalid inputs, returning False")
+        return False, ret
+
+    log.info("checking stats of {0}".format(filepath))
+    if os.path.exists(filepath):
+        salt_ret = __salt__['file.stats'](filepath)
+    else:
+        salt_ret = {}
+
+    log.debug("file stats are {0}".format(salt_ret))
+    if not salt_ret:
+        log.info("file stats couldn't be fetched, checking corner cases")
+        return _check_corner_cases(filepath, expected)
+
+    passed = True
+    reason_dict = {}
+    for attribute in expected.keys():
+        if attribute == 'allow_more_strict' or attribute == 'match_on_file_missing':
+            continue
+        file_attribute_value = salt_ret[attribute]
+
+        if attribute == 'mode':
+            if file_attribute_value != '0':
+                file_attribute_value = file_attribute_value[1:]
+            allow_more_strict = False
+            if 'allow_more_strict' in expected.keys():
+                allow_more_strict = expected['allow_more_strict']
```

Comment:
can be combined in a single line
`allow_more_strict = expected.get('allow_more_strict', False)`

Reply:
Done.